scripts/bulk-optimizer.js

```diff
@@ -0,0 +1,269 @@
+            padding: 2rem;
+            border-radius: 8px;
+            text-align: center;
+            width: 400px;
+            box-shadow: 0 0 15px rgba(0, 0, 0, 0.2);
+        `;
+
+		const modalTitle = document.createElement( 'h2' );
+		modalTitle.id = 'nfd-modal-title';
+		modalTitle.textContent = 'Optimizing Images...';
+
+		const currentFileName = document.createElement( 'p' );
+		currentFileName.id = 'nfd-current-file';
+		currentFileName.textContent = 'Preparing files...';
+
+		const progressContainer = document.createElement( 'div' );
+		progressContainer.style.cssText = `
+            width: 100%;
+            height: 20px;
+            background: #eee;
+            border-radius: 10px;
+            margin: 1rem 0;
+            overflow: hidden;
+            position: relative;
+        `;
+
+		const progressBar = document.createElement( 'div' );
+		progressBar.id = 'nfd-progress-bar';
+		progressBar.style.cssText = `
+            height: 100%;
+            width: 0;
+            background: #007cba;
+            transition: width 0.3s ease;
+        `;
+
+		const cancelButton = document.createElement( 'button' );
+		cancelButton.textContent = 'Cancel';
+		cancelButton.className = 'button button-secondary';
+		cancelButton.style.marginTop = '1rem';
+		cancelButton.addEventListener( 'click', () => {
+			cancelRequested = true;
+			closeModal();
+			window.location.reload(); // Reload on cancel
+		} );
+
+		progressContainer.appendChild( progressBar );
+		modalContent.append(
+			modalTitle,
+			currentFileName,
+			progressContainer,
+			cancelButton
+		);
+		modal.appendChild( modalContent );
+		document.body.appendChild( modal );
+
+		return { modal, progressBar, modalTitle, currentFileName };
+	};
+
+	/**
+	 * Opens the modal and resets its display.
+	 */
+	const openModal = () => {
+		cancelRequested = false;
+		const { progressBar, modalTitle, currentFileName } = createModal();
+		progressBar.style.width = '0%';
+		currentFileName.textContent = '';
+		return { progressBar, modalTitle, currentFileName };
+	};
+
+	/**
+	 * Closes the modal and reloads the page.
+	 */
+	const closeModal = () => {
+		const modal = document.getElementById( 'nfd-bulk-modal' );
+		if ( modal ) modal.remove();
+		window.location.reload(); // Reload on close
+	};
+
+	/**
+	 * Extracts the file name from the attachment element.
+	 *
+	 * @param attachment
+	 */
+	const getFileName = ( attachment ) => {
+		const mediaContent = attachment.closest( '.media-frame-content' );
+		const fileNameElement = mediaContent?.querySelector( '.filename' );
+		return fileNameElement?.textContent || 'Unknown File';
+	};
+
+	/**
+	 * Handles bulk optimization with progress bar and filename display.
+	 */
+	const handleBulkOptimize = async () => {
+		const selectedItems = Array.from(
+			document.querySelectorAll( '.attachment.selected' )
+		).map( ( attachment ) => ( {
+			id: attachment.getAttribute( 'data-id' ),
+			name: getFileName( attachment ),
```

Comment:
it seems filename is always the same if we have more than one image selected because in getFileName function on 134 line where the file name is get by the .media-frame-content .filename that only provide the current selected file, so in case of more selected we only get the last seletec element file name.

 Maybe we could change the line  like this

```suggestion
			name: attachment.getAttribute( 'aria-label' ),
```
and then remove the getFileName function on line 133

Reply:
Thanks! @geckod22 updated.